### ISSUE TITLE
SwiftUI: Images always flicker and shows the placeholder briefly even if cached

### ISSUE THREAD

**Issue by hyouuu** [onevcat/Kingfisher#1549]
### Check List

Thanks for considering to open an issue. Before you submit your issue, please confirm these boxes are checked.

- [v] I have read the [wiki page](https://github.com/onevcat/Kingfisher/wiki) and [cheat sheet](https://github.com/onevcat/Kingfisher/wiki/Cheat-Sheet), but there is no information I need.
- [v] I have searched in [existing issues](https://github.com/onevcat/Kingfisher/issues?utf8=✓&q=is%3Aissue), but did not find a same one.
- [v] I want to report a problem instead of asking a question. It'd better to use [kingfisher tag in Stack Overflow](http://stackoverflow.com/questions/tagged/kingfisher) to ask a question.

#### What

SwiftUI: Images always flicker and show the placeholder briefly even if cached

#### Reproduce

```
KFImage(source: story.kfImageSource())
                            .onSuccess { r in
                                print("kf suc: \(r)")
                                pr()
                            }
                            .onFailure { e in
                                print("kf err: \(e)")
                                pr()
                            }
                            .placeholder {
                                Image(uiImage: Themer.loadImage("cover", overrideColor: .gray))
                                    .side(side)
                                    .cornerRadius(5)
                            }
                            .side(side)
                            .cornerRadius(5)
```

Where kfImageSource is:
```
func kfImageSource() -> Kingfisher.Source? {
        guard let url = imageDownloadURL() else {
            #if DEBUG
            var name = ""
            if let user = self as? User {
                name = user.name
            } else if let story = self as? Story {
                name = story.title
            }
            if let imageKey = imageKey {
                pr("kfImageSource nil AWS preSignedURL for imageKey:\(imageKey) name:\(name)")
            } else {
                pr("kfImageSource nil imageKey name:\(name)")
            }
            #endif
            return nil
        }
        let imageResource = ImageResource(downloadURL: url, cacheKey: imageKey)
        return Kingfisher.Source.network(imageResource)
    }
```

onFailure never get called so everything works normally, and manually verified cache is there.

#### Other Comment

It didn't happen before so I looked at recent KF releases and the 5.15.1 release caught my eye - https://github.com/onevcat/Kingfisher/pull/1510/files moved image assigning to onAppear instead of init. So I tried to downgrade to 5.15.0 and the problem is gone.

**Comment by onevcat** [comment]
Should be fixed in 5.15.6.

**Comment by hyouuu** [comment]
Confirmed fixed - thanks!
